lib/dry/schema/macros/value.rb

```diff
@@ -38,6 +38,16 @@ def call(*predicates, **opts, &block)
 
             if block && type_spec.equal?(:hash)
               hash(&block)
+            elsif type_spec.respond_to?(:keys)
+              hash do
+                type_spec.each do |key|
+                  if key.required?
+                    required(key.name).value(key.type)
+                  else
+                    optional(key.name).value(key.type)
+                  end
+                end
+              end
```

Comment:
@flash-gordon we could tweak `hash` to accept a hash schema so that this becomes `hash(type_spec)`

Reply:
sure, thanks for the hint. I actually have no plans leave it as such, just making it work first